### PR TITLE
Fixes the release `400` error on smoke tests

### PR DESCRIPTION
**Pull request by crdant**
TL;DR
-----

Makes sure versions are aligned when running smoke tests

Details
-------

Replaces hardcode major.minor.patch of `0.0.1` from the version used in the smoke action and replaces it with the version from the Helm chart's `Chart.yaml`. I thought that would be all that was required to fix the `400` on release, but it wasn't. So now the smoke test also updates `Chart.yaml` with the version that it creates to use for the test release. This required by the strict rules the vendor portal has for validating that the release version matches the chart version.

Ideally we could stash the build related tagging into the build metadata fragment of the semver instead of the prerelease version, and the vendor portal could ignore build metadata.

### CODE REVIEW

File: smoke-test/action.yml [hunk 1]
```diff
@@ -13,6 +13,9 @@ inputs:
     description: 'The directory containing multiple yamls for a Replicated release.'
     required: true
     default: 'manifests'
+  version:
+    description: 'Provides version to use for the Replicated Release'
+    reqired: false
```
Comment (jdewinne) [inline smoke-test/action.yml:18]:
```suggestion
    required: false
```

File: smoke-test/action.yml [hunk 2]
```diff
@@ -39,12 +42,59 @@ inputs:
 runs:
   using: "composite"
   steps:
+  - name: Extract Chart YAML
+    id: extract-chart-yaml
+    if: inputs.version == ''
+    shell: bash
+    run: |
+      tarball=$(find ${YAML_DIR} -name "${CHART_NAME}*.tgz")
```
Comment (jdewinne) [inline smoke-test/action.yml:50]: I guess this will break if the `yaml-dir` doesn't contain a helm chart?

File: smoke-test/action.yml [hunk 3]
```diff
@@ -39,12 +42,59 @@ inputs:
...
+    uses: mikefarah/yq@master
+    with:
+      cmd: yq '.version' ${CHART_YAML}
   - name: Define App Version
     shell: bash
-    run: echo "APP_VERSION=0.0.1-${GITHUB_REF_NAME//[^a-zA-Z0-9]/}.${GITHUB_RUN_ID}${GITHUB_RUN_ATTEMPT}" >> $GITHUB_ENV
+    if: inputs.version == ''
+    run: echo "APP_VERSION=${CHART_VERSION}+${GITHUB_REF_NAME//[^a-zA-Z0-9]/}.${GITHUB_RUN_ID}${GITHUB_RUN_ATTEMPT}" >> $GITHUB_ENV
+    env:
+      CHART_VERSION: ${{ steps.get-chart-version.outputs.result }}
```
Comment (jdewinne) [inline smoke-test/action.yml:67]: Other idea: If `installation-method` is `helm`, extract the version from `Chart.yaml`. Doing that would also remove the extra requirement to specify an input version.

**Review by jdewinne — CHANGES_REQUESTED** [review]
I would do this different, and in the case of `helm`, automatically extract the app version from the Chart.yaml, vs having it dynamic.

File: smoke-test/action.yml (hunk 3 above)
Comment (crdant) [reply smoke-test/action.yml:67]: It's actually an either/or. I just flubbed making the version required. It will use the one you provide, or extract it and update it to a build specific version number.

File: smoke-test/action.yml (hunk 2 above)
Comment (crdant) [reply smoke-test/action.yml:50]: yes, all the manipulation of the chart will break, I should make those conditional

Comment (crdant) [reply smoke-test/action.yml:50]: made a few more things conditional

File: smoke-test/action.yml (hunk 3 above)
Comment (crdant) [reply smoke-test/action.yml:67]: yes, extended out. the conditional

**Comment by xavpaice** [comment]
@crdant this has been a while, is it still needing review or is it no longer required?

**Comment by crdant** [comment]
> @crdant this has been a while, is it still needing review or is it no longer required?

I've generally moved away from our actions so I'm not sure. Happy to check it again.
